pkg/api/types.go

```diff
@@ -360,6 +360,8 @@ type KubernetesConfig struct {
 	LoadBalancerSku                  string            `json:"loadBalancerSku,omitempty"`
 	ExcludeMasterFromStandardLB      *bool             `json:"excludeMasterFromStandardLB,omitempty"`
 	AzureCNIVersion                  string            `json:"azureCNIVersion,omitempty"`
+	AzureCNIURLLinux                 string            `json:"azureCNIURLLinux,omitempty"`
```

Comment:
Can we have a struct (AzureCNIConfig perhaps? )that wraps AzureCNIVersion, AzureCIURLLinux, AzureCNIURLWindows, and AzureCNIPluginURL ? This way we make all the four user configurable and struct can be reused in KubernetesConfig as well as KubernetesSpecConfig structs.

KubernetesConfig is getting really bloated right now :(